### DISCUSSION TITLE
tree shaking

### DISCUSSION THREAD

**Discussion by JLarky** [alephjs/aleph.js#80] [General] [answered]
I'm trying to understand why tree shaking is not working in aleph :)

```
import {myFunc} from "./myModule.ts";
const x = useDeno(() => myFunc());
```

I'm would expect if I only use `myFunc` on server why do I need to get all of that downloaded to my client? :) If `myModule.ts` depends on markdown parser or something like that, all of this is getting sent to client :) I can create sample example if my point is not clear, but I'm curios to just discuss this topic and if there are any plans to address this, maybe https://github.com/alephjs/react-io :)

**Comment by ije** [comment, marked as answer]
currently, the tsc just transpiles the code, the bundler i'm working on it will support the tree shaking by swc.

**Reply by shadowtime2000** [reply]
The bundler is being created in #28
